[regression] 0.9.6 crashes when clicking "Example" as a first action

```shell
$ flatpak install --user https://dl.flathub.org/build-repo/74697/cz.cvut.edu.comparch.qtrvsim.flatpakref
$ flatpak run cz.cvut.edu.comparch.qtrvsim
```

When you open the application and first action is clicking "Example" button, application silently crashes.

Works for 0.9.5 or when you press something else before pressing "Example" button

With the help of bitect, I found out that the commit causing the problem is https://github.com/cvut/qtrvsim/commit/16c78c16948799a008b47569ea53989623c33e1f .

<details>

<summary>[Bitect details](https://github.com/cvut/qtrvsim/commit/16c78c16948799a008b47569ea53989623c33e1f)</summary>

```text
16c78c16948799a008b47569ea53989623c33e1f is the first bad commit
commit 16c78c16948799a008b47569ea53989623c33e1f
Author: Jakub Dupak <dev@jakubdupak.com>
Date:   Sun Jul 16 23:40:44 2023 +0200

    GUI: split central widget tabs to coreview and editor

 src/gui/mainwindow/mainwindow.cpp | 17 +++++++++++------
 src/gui/mainwindow/mainwindow.h   |  3 ++-
 2 files changed, 13 insertions(+), 7 deletions(-)
```

</details>

I have checked https://github.com/cvut/qtrvsim/commit/3dcd6963818c0f368d6ae5713d3b1b14f4d98217 whose everything's fine.

I have checked https://github.com/cvut/qtrvsim/commit/16c78c16948799a008b47569ea53989623c33e1f which causes an error during building process.

<details>

<summary>Compile Log</summary>

```bash
PS E:\56\qtrvsim> D:\CMake\bin\cmake.exe E:\56\qtrvsim -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_C_COMPILER_LAUNCHER="E:\ccache-4.9-windows-x86_64\ccache.exe" -DCMAKE_CXX_COMPILER_LAUNCHER="E:\ccache-4.9-windows-x86_64\ccache.exe" -DFORCE_COLORED_OUTPUT=true -G "MinGW Makefiles"
-- Debug prints globally suppressed.
-- Could NOT find LibElf (missing: LIBELF_LIBRARY LIBELF_INCLUDE_DIR)
-- Using local libelf fallback.
-- Qt5 version: 5.12.9
-- Qt5 print support: 1
-- Configuring done (0.2s)
-- Generating done (0.2s)
-- Build files have been written to: E:/56/qtrvsim
PS E:\56\qtrvsim> D:\CMake\bin\cmake.exe --build . -j24
[  2%] Automatic MOC for target mulh64_test
[  3%] Automatic MOC and UIC for target svgscene
[  3%] Automatic MOC for target registers_test
[  5%] [  5%] Automatic MOC for target alu_test
Automatic MOC for target memory_test
[  5%] Automatic MOC for target assembler
[  6%] [  6%] Automatic MOC for target cache_test
Automatic MOC for target os_emulation
[ 30%] Built target libelf
[ 30%] Built target svgscene_autogen
[ 30%] Built target instruction_test_autogen
[ 30%] Built target mulh64_test_autogen
[ 30%] Built target alu_test_autogen
[ 30%] Built target registers_test_autogen
[ 30%] Built target memory_test_autogen
[ 30%] Built target cache_test_autogen
[ 30%] Built target assembler_autogen
[ 30%] Built target os_emulation_autogen
[ 31%] Automatic MOC for target core_test
[ 31%] Automatic MOC for target machine
[ 32%] Automatic MOC for target program_loader_test
[ 32%] Built target core_test_autogen
[ 32%] Built target machine_autogen
[ 32%] Built target program_loader_test_autogen
[ 33%] Built target mulh64_test
[ 34%] Built target os_emulation
[ 36%] Built target assembler
[ 41%] Built target svgscene
[ 42%] Built target alu_test
[ 44%] Built target registers_test
[ 47%] Built target memory_test
[ 53%] Built target cache_test
[ 55%] Built target instruction_test
[ 58%] Built target program_loader_test
[ 68%] Built target machine
[ 74%] Built target core_test
[ 75%] Automatic MOC for target cli
[ 76%] Automatic MOC and UIC for target gui
[ 76%] Built target cli_autogen
[ 79%] Built target cli
[ 79%] Built target gui_autogen
[ 80%] Building CXX object src/gui/CMakeFiles/gui.dir/gui_autogen/mocs_compilation.cpp.obj
[ 81%] Building CXX object src/gui/CMakeFiles/gui.dir/dialogs/new/newdialog.cpp.obj
[ 82%] Building CXX object src/gui/CMakeFiles/gui.dir/main.cpp.obj
[ 82%] Building CXX object src/gui/CMakeFiles/gui.dir/mainwindow/mainwindow.cpp.obj
[ 82%] Linking CXX executable ..\..\target\qtrvsim_gui.exe
[100%] Built target gui
PS E:\56\qtrvsim> D:\CMake\bin\cmake.exe E:\56\qtrvsim -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_C_COMPILER_LAUNCHER="E:\ccache-4.9-windows-x86_64\ccache.exe" -DCMAKE_CXX_COMPILER_LAUNCHER="E:\ccache-4.9-windows-x86_64\ccache.exe" -DFORCE_COLORED_OUTPUT=true -G "MinGW Makefiles"
-- Debug prints globally suppressed.
-- Could NOT find LibElf (missing: LIBELF_LIBRARY LIBELF_INCLUDE_DIR)
-- Using local libelf fallback.
-- Qt5 version: 5.12.9
-- Qt5 print support: 1
-- Configuring done (0.2s)
-- Generating done (0.2s)
-- Build files have been written to: E:/56/qtrvsim
PS E:\56\qtrvsim> D:\CMake\bin\cmake.exe --build . -j24
[  1%] Automatic MOC and UIC for target svgscene
[  2%] Automatic MOC for target mulh64_test
[  2%] Automatic MOC for target registers_test
[  3%] Automatic MOC for target memory_test
[  4%] Automatic MOC for target alu_test
[  4%] Automatic MOC for target assembler
[  4%] Automatic MOC for target cache_test
[  5%] Automatic MOC for target instruction_test
[  6%] Automatic MOC for target os_emulation
[ 30%] Built target libelf
[ 30%] Built target svgscene_autogen
[ 30%] Built target mulh64_test_autogen
[ 30%] Built target registers_test_autogen
[ 30%] Built target memory_test_autogen
[ 30%] Built target alu_test_autogen
[ 30%] Built target cache_test_autogen
[ 30%] Built target instruction_test_autogen
[ 30%] Built target assembler_autogen
[ 30%] Built target os_emulation_autogen
[ 30%] Automatic MOC for target machine
[ 31%] Automatic MOC for target core_test
[ 32%] Automatic MOC for target program_loader_test
[ 32%] Built target machine_autogen
[ 32%] Built target program_loader_test_autogen
[ 32%] Built target core_test_autogen
[ 33%] Built target mulh64_test
[ 34%] Built target os_emulation
[ 35%] Built target alu_test
[ 40%] Built target svgscene
[ 42%] Built target registers_test
[ 44%] Built target assembler
[ 47%] Built target memory_test
[ 53%] Built target cache_test
[ 55%] Built target instruction_test
[ 58%] Built target program_loader_test
[ 68%] Built target machine
[ 74%] Built target core_test
[ 76%] [ 76%] Automatic MOC for target cli
Automatic MOC and UIC for target gui
[ 76%] Built target cli_autogen
[ 79%] Built target cli
[ 79%] Built target gui_autogen
[ 80%] Building CXX object src/gui/CMakeFiles/gui.dir/gui_autogen/mocs_compilation.cpp.obj
[ 81%] Building CXX object src/gui/CMakeFiles/gui.dir/dialogs/new/newdialog.cpp.obj
[ 82%] Building CXX object src/gui/CMakeFiles/gui.dir/main.cpp.obj
[ 82%] Building CXX object src/gui/CMakeFiles/gui.dir/mainwindow/mainwindow.cpp.obj
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp: In constructor 'MainWindow::MainWindow(QSettings*, QWidget*)':
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:56:26: error: 'class HidingTabWidget' has no member named 'setTabVisibl
'; did you mean 'setVisible'?
     central_widget_tabs->setTabVisible(central_widget_tabs->indexOf(editor_tabs.data()), false);
                          ^~~~~~~~~~~~~
                          setVisible
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:141:9: error: 'central_window' was not declared in this scope
         central_window.data(), &HidingTabWidget::currentChanged, this,
         ^~~~~~~~~~~~~~
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:141:9: note: suggested alternative: 'centralWidget'
         central_window.data(), &HidingTabWidget::currentChanged, this,
         ^~~~~~~~~~~~~~
         centralWidget
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp: In member function 'void MainWindow::show_hide_coreview(bool)':
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:180:13: error: 'central_window' was not declared in this scope
             central_window->removeTab(central_window->indexOf(coreview.data()));
             ^~~~~~~~~~~~~~
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:180:13: note: suggested alternative: 'centralWidget'
             central_window->removeTab(central_window->indexOf(coreview.data()));
             ^~~~~~~~~~~~~~
             centralWidget
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:194:5: error: 'central_window' was not declared in this scope
     central_window->insertTab(0, coreview.data(), "Core");
     ^~~~~~~~~~~~~~
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:194:5: note: suggested alternative: 'centralWidget'
     central_window->insertTab(0, coreview.data(), "Core");
     ^~~~~~~~~~~~~~
     centralWidget
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp: In member function 'void MainWindow::create_core(const machine::MachineConfig&, bool, bool)':
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:233:26: error: 'central_window' was not declared in this scope
     auto focused_index = central_window->currentIndex();
                          ^~~~~~~~~~~~~~
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:233:26: note: suggested alternative: 'centralWidget'
     auto focused_index = central_window->currentIndex();
                          ^~~~~~~~~~~~~~
                          centralWidget
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp: In member function 'void MainWindow::save_exit_or_ignore(bool, const QStringList&)':
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:472:26: error: 'central_window' was not declared in this scope
     if (save_unnamed && (central_window != nullptr)) {
                          ^~~~~~~~~~~~~~
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:472:26: note: suggested alternative: 'centralWidget'
     if (save_unnamed && (central_window != nullptr)) {
                          ^~~~~~~~~~~~~~
                          centralWidget
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp: In member function 'void MainWindow::central_tab_changed(int)':
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:560:23: error: 'central_window' was not declared in this scope
     QWidget *widget = central_window->widget(index);
                       ^~~~~~~~~~~~~~
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:560:23: note: suggested alternative: 'centralWidget'
     QWidget *widget = central_window->widget(index);
                       ^~~~~~~~~~~~~~
                       centralWidget
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp: In member function 'void MainWindow::add_src_editor_to_tabs(SrcEditor*
':
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:566:5: error: 'central_window' was not declared in this scope
     central_window->addTab(editor, editor->title());
     ^~~~~~~~~~~~~~
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:566:5: note: suggested alternative: 'centralWidget'
     central_window->addTab(editor, editor->title());
     ^~~~~~~~~~~~~~
     centralWidget
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp: In member function 'void MainWindow::update_open_file_list()':
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:573:10: error: 'central_window' was not declared in this scope
     if ((central_window == nullptr) || (settings == nullptr)) { return; }
          ^~~~~~~~~~~~~~
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:573:10: note: suggested alternative: 'centralWidget'
     if ((central_window == nullptr) || (settings == nullptr)) { return; }
          ^~~~~~~~~~~~~~
          centralWidget
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:574:25: error: 'central_window' was not declared in this scope
     for (int i = 0; i < central_window->count(); i++) {
                         ^~~~~~~~~~~~~~
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:574:25: note: suggested alternative: 'centralWidget'
     for (int i = 0; i < central_window->count(); i++) {
                         ^~~~~~~~~~~~~~
                         centralWidget
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp: In member function 'bool MainWindow::modified_file_list(QStringList&, bool)':
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:587:9: error: 'central_window' was not declared in this scope
     if (central_window == nullptr) { return false; }
         ^~~~~~~~~~~~~~
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:587:9: note: suggested alternative: 'centralWidget'
     if (central_window == nullptr) { return false; }
         ^~~~~~~~~~~~~~
         centralWidget
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:588:25: error: 'central_window' was not declared in this scope
     for (int i = 0; i < central_window->count(); i++) {
                         ^~~~~~~~~~~~~~
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:588:25: note: suggested alternative: 'centralWidget'
     for (int i = 0; i < central_window->count(); i++) {
                         ^~~~~~~~~~~~~~
                         centralWidget
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp: In member function 'SrcEditor* MainWindow::source_editor_for_file(const QString&, bool)':
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:611:9: error: 'central_window' was not declared in this scope
     if (central_window == nullptr) { return nullptr; }
         ^~~~~~~~~~~~~~
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:611:9: note: suggested alternative: 'centralWidget'
     if (central_window == nullptr) { return nullptr; }
         ^~~~~~~~~~~~~~
         centralWidget
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:614:25: error: 'central_window' was not declared in this scope
     for (int i = 0; i < central_window->count(); i++) {
                         ^~~~~~~~~~~~~~
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:614:25: note: suggested alternative: 'centralWidget'
     for (int i = 0; i < central_window->count(); i++) {
                         ^~~~~~~~~~~~~~
                         centralWidget
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp: In member function 'void MainWindow::open_source()':
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:652:17: error: 'central_window' was not declared in this scope
             if (central_window != nullptr) { central_window->setCurrentWidget(editor); }
                 ^~~~~~~~~~~~~~
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:652:17: note: suggested alternative: 'centralWidget'
             if (central_window != nullptr) { central_window->setCurrentWidget(editor); }
                 ^~~~~~~~~~~~~~
                 centralWidget
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp: In member function 'void MainWindow::save_source_as()':
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:687:15: error: 'central_window' was not declared in this scope
     int idx = central_window->indexOf(current_srceditor);
               ^~~~~~~~~~~~~~
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:687:15: note: suggested alternative: 'centralWidget'
     int idx = central_window->indexOf(current_srceditor);
               ^~~~~~~~~~~~~~
               centralWidget
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp: In member function 'void MainWindow::src_editor_save_to(const QString&
':
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:710:15: error: 'central_window' was not declared in this scope
     int idx = central_window->indexOf(current_srceditor);
               ^~~~~~~~~~~~~~
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:710:15: note: suggested alternative: 'centralWidget'
     int idx = central_window->indexOf(current_srceditor);
               ^~~~~~~~~~~~~~
               centralWidget
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp: In member function 'void MainWindow::close_source()':
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:770:15: error: 'central_window' was not declared in this scope
     int idx = central_window->indexOf(editor);
               ^~~~~~~~~~~~~~
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:770:15: note: suggested alternative: 'centralWidget'
     int idx = central_window->indexOf(editor);
               ^~~~~~~~~~~~~~
               centralWidget
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp: In member function 'void MainWindow::message_selected(messagetype::Type, const QString&, int, int, const QString&, const QString&)':
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:819:9: error: 'central_window' was not declared in this scope
     if (central_window != nullptr) { central_window->setCurrentWidget(editor); }
         ^~~~~~~~~~~~~~
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:819:9: note: suggested alternative: 'centralWidget'
     if (central_window != nullptr) { central_window->setCurrentWidget(editor); }
         ^~~~~~~~~~~~~~
         centralWidget
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp: In member function 'virtual bool SimpleAsmWithEditorCheck::process_pragma(QStringList&, const QString&, int, QString*)':
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:867:29: error: 'class MainWindow' has no member named 'central_window'; did you mean 'centralWidget'?
             && (mainwindow->central_window != nullptr) && (mainwindow->coreview != nullptr)) {
                             ^~~~~~~~~~~~~~
                             centralWidget
E:\56\qtrvsim\src\gui\mainwindow\mainwindow.cpp:868:25: error: 'class MainWindow' has no member named 'central_window'; did you mean 'centralWidget'?
             mainwindow->central_window->setCurrentWidget(mainwindow->coreview.data());
                         ^~~~~~~~~~~~~~
                         centralWidget
mingw32-make.exe[2]: *** [src\gui\CMakeFiles\gui.dir\build.make:477: src/gui/CMakeFiles/gui.dir/mainwindow/mainwindow.cpp.obj] Error 1
mingw32-make.exe[1]: *** [CMakeFiles\Makefile2:1068: src/gui/CMakeFiles/gui.dir/all] Error 2
mingw32-make.exe: *** [Makefile:145: all] Error 2
```

</details>

## Environment Info

<details>

<summary>Configurations</summary>

```text
-- The C compiler identification is GNU 8.1.0
-- The CXX compiler identification is GNU 8.1.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: E:/mingw64/bin/gcc.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: E:/mingw64/bin/g++.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Debug prints globally suppressed.
-- Could NOT find LibElf (missing: LIBELF_LIBRARY LIBELF_INCLUDE_DIR)
-- Using local libelf fallback.
-- Qt5 version: 5.12.9
-- Qt5 print support: 1
-- Configuring done (1.6s)
-- Generating done (0.2s)
-- Build files have been written to: E:/56/qtrvsim
```

</details>

Qt 5.12.9 comes from [Qt Downloads](https://download.qt.io/)](https://download.qt.io/archive/qt/5.12/5.12.9/qt-opensource-windows-x86-5.12.9.exe).

Compiler comes from [mingw-w64](https://sourceforge.net/projects/mingw-w64/files/Toolchains%20targetting%20Win64/Personal%20Builds/mingw-builds/8.1.0/threads-win32/seh/x86_64-8.1.0-release-win32-seh-rt_v6-rev0.7z).

--- a/src/gui/windows/editor/editordock.cpp
+++ b/src/gui/windows/editor/editordock.cpp
@@ -48,7 +48,7 @@ EditorDock::EditorDock(QSharedPointer<QSettings> settings, QTabWidget *parent_ta
         this, &EditorDock::currentChanged, parent_tabs,
         [this, parent_tabs](int index) {
             // Update parent title
-            if (count() == 0) return;
+            if (count() == 0 || index < 0) return;
             auto *editor = get_tab(index)->get_editor();
             QString title = QString("&Editor (%1)").arg(editor->title());
             parent_tabs->setTabText(parent_tabs->indexOf(this), title);

